Library/Homebrew/utils/git.rb

```diff
@@ -14,6 +20,31 @@ def last_revision_commit_of_file(repo, file, before_commit: nil)
     out.chomp
   end
 
+  def last_revision_commit_of_files(repo, files, before_commit: nil)
```

Comment:
This function feels like it has a lot of overlap. Could it share some code with the previous function or be one function that takes different parameters? Also, it may be worth just living in `extract.rb` until it has more than one caller.

Reply:
I think it's better to leave them as separate functions. They both call `git log` with different arguments but these arguments are set to make sure `git log` output is of a certain format. The two functions are functionally different too. They expect different things, look for different things and return different things. They do look kinda similar now but I've gone through versions of `last_revision_commit_of_files` where it looks very different from `last_revision_commit_of_file`. In the future if there's a need to update `last_revision_commit_of_files` it would be easier because it's not constrained by the other one and vice versa.

Reply:
Also I think should leave it here with the other functions coz they're related.

Reply:
I'm unsure about this being a "utils" function when it's used only once. I could see the argument for keeping them alongside each other if they shared code but they don't. Your thoughts?

Reply:
My argument for keeping it in the same place with the other two functions is that they are kinda close in terms of what they do. But I'm happy either way.
I notice that the other 2 functions in `utils/git` are also only used in `extract` 🤔

Reply:
🆒 👍